assets/product-info.js

```diff
@@ -77,12 +68,14 @@ if (!customElements.get('product-info')) {
         const productUrl = target.dataset.productUrl || this.pendingRequestUrl || this.dataset.url;
         this.pendingRequestUrl = productUrl;
         const shouldSwapProduct = this.dataset.url !== productUrl;
-        const shouldFetchFullPage = !this.isFeaturedProduct && shouldSwapProduct;
+        const shouldFetchFullPage = this.dataset.updateUrl === 'true' && shouldSwapProduct;
```

Comment:
Prior to this change, we were requesting a full PDP from both main-product and quick add modal, and just requesting the section for featured product. However, for the quick add modal we only need the main product section.

This change makes it so that we request the full page when swapping between combined listings in a main product section, and request just the product section for both featured product and quick add modal.